### ISSUE TITLE
infinite loop in MonarchLexer's getloadstatus

### ISSUE THREAD

**Issue by corrieann** [microsoft/vscode#76753]
- VSCode Version: n/a -- Using Monaco 
- OS Version: MacOS Mojave

Steps to Reproduce:

1. Create a language syntax that references itself as an embedded mode.
2. Call colorizeElement using that language for syntax highlighting.

Does this issue occur when all extensions are disabled?: Yes

Code in question is:
https://github.com/microsoft/vscode/blob/master/src/vs/editor/standalone/common/monarch/monarchLexer.ts#L423

Proposed fix is:
```
if (tokenizationSupport instanceof MonarchTokenizer) {
  const nestedModeStatus = tokenizationSupport.getLoadStatus();
  ...

```

become:
```
if (tokenizationSupport == this) continue;
if (tokenizationSupport instanceof MonarchTokenizer) {
  const nestedModeStatus = tokenizationSupport.getLoadStatus();
  ...
```

**Comment by fabiospampinato** [comment]
I've just stumbled upon this too. 

In case it's useful I've noticed that the call to colorize/colorizeElement works at first if I'm just viewing the rendered Markdown, which is the language that references itself as an embedded mode for me, but then if I fire up a monaco editor instance with that same Markdown content and switch back to rendering and syntax highlighting it than the bug occurs. It doesn't happen if I start editing some Markdown which doesn't embed a markdown code block inside itself and then switch back to the Markdown I was rendering initially.
